src/node/blockstorage.cpp

```diff
@@ -397,6 +404,16 @@ bool BlockManager::IsBlockPruned(const CBlockIndex* pblockindex)
     return (m_have_pruned && !(pblockindex->nStatus & BLOCK_HAVE_DATA) && pblockindex->nTx > 0);
 }
 
+const CBlockIndex* GetFirstStoredBlock(const CBlockIndex* start_block) {
+    AssertLockHeld(::cs_main);
+    assert(start_block);
```

Comment:
Why is this function accepting a nullptr, but then aborting on it. Wouldn't it be better to not accept a nullptr (aka reference)?

Reply:
> Why is this function accepting a nullptr, but then aborting on it. Wouldn't it be better to not accept a nullptr (aka reference)?

Proposed in #24150.